datafusion/core/src/execution/context.rs

```diff
@@ -1561,8 +1561,7 @@ impl ContextProvider for SessionState {
             return None;
         }
 
-        let first_variable = &variable_names[0];
-        let provider_type = if first_variable.len() > 1 && &first_variable[0..2] == "@@" {
+        let provider_type = if is_system_variables(variable_names) {
```

Comment:
This is a really nice idea to extract the common logic into a function and unit test it 🏅